sbom-utility-scripts/scripts/index-image-sbom-script/index_image_sbom_script.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python3
+import argparse
+import hashlib
+import json
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Optional, Any
+from dataclasses import dataclass
+from uuid import uuid4
+
+from packageurl import PackageURL
+
+
+@dataclass
+class Image:
+    repository: str
+    name: str
+    digest: str
+    tag: str
+    arch: Optional[str]
+
+    @staticmethod
+    def from_image_index_url_and_digest(
+        image_url_and_tag: str,
+        image_digest: str,
+        arch: Optional[str] = None,
+    ) -> "Image":
+
+        repository, tag = image_url_and_tag.rsplit(":", 1)
+        _, name = repository.rsplit("/", 1)
+        return Image(
+            repository=repository,
+            name=name,
+            digest=image_digest,
+            tag=tag,
+            arch=arch,
+        )
+
+    @property
+    def digest_algo(self) -> str:
+        algo, _ = self.digest.split(":")
+        return algo.upper()
+
+    @property
+    def digest_hex_val(self) -> str:
+        _, val = self.digest.split(":")
+        return val
+
+    def purls(self, index_digest: Optional[str] = None) -> list[str]:
+        ans = []
+        if index_digest and self.arch:
+            ans.append(
+                PackageURL(
+                    type="oci",
+                    name=self.name,
+                    version=index_digest,
+                    qualifiers={"arch": self.arch, "repository_url": self.repository},
+                ).to_string()
+            )
+        ans.append(
+            PackageURL(
+                type="oci", name=self.name, version=self.digest, qualifiers={"repository_url": self.repository}
+            ).to_string()
+        )
+        return ans
+
+    def propose_spdx_id(self) -> str:
+        purl_hex_digest = hashlib.sha256(self.purls()[0].encode()).hexdigest()
+        return f"SPDXRef-Image-{self.name}-{purl_hex_digest}"
```

Comment:
nitpick: if we want to include the type in the SPDXID, let's be consistent about lowercase/uppercase (`Image` vs. `image-index`)

Alternatively, maybe we can just stick to `SPDXRef-{name}-{purl_digest}` (for everything, the image index and the images). It looks like your PR will be going in first, so we can set the precedent here :slightly_smiling_face:

Reply:
The second point is an interesting idea. However I think it might slightly decrease the human-readability of the SBOM as the relationships will be less clear at a first sight. If you don't protest against this, I think I will just make the `Image` lowercase.

Reply:
works for me :+1: